ipie/addons/thermal/estimators/handler.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2022 The ipie Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Fionn Malone <fmalone@google.com>
+#          Joonho Lee
+#
+
+"""Routines and classes for estimation of observables."""
+
+import os
+from typing import Tuple, Union
+
+import h5py
+import numpy
+
+from ipie.config import config, MPI
+from ipie.addons.thermal.estimators.energy import ThermalEnergyEstimator
+from ipie.addons.thermal.estimators.particle_number import ThermalNumberEstimator
+from ipie.estimators.estimator_base import EstimatorBase
+from ipie.estimators.utils import H5EstimatorHelper
+from ipie.utils.io import format_fixed_width_strings
+
+# Some supported (non-custom) estimators
+_predefined_estimators = {
+    "energy": ThermalEnergyEstimator,
+    "nav": ThermalNumberEstimator,
+}
+
+
+class ThermalEstimatorHandler(object):
```

Comment:
From a glance I think this is essentialy identical to EstimatorHandler, with the only specialization being the print_time_slice method. You can use inheritance and reduce the amount of code here.

Reply:
The `thermal/estimators/` folder has no dependence on the non-thermal `systems` classes, so I'll just keep the methods that don't call `systems` but do in the non-thermal code.